Description: CVE-2024-1013 - Fix incompatible pointer-to-integer types in PostgreSQL driver
 Replace SDWORD and Int4 (32-bit) with SQLLEN (64-bit compatible) to fix
 out-of-bounds stack write vulnerability on 64-bit architectures where
 the caller has 4 bytes and callee writes 8 bytes.
Origin: upstream, https://github.com/lurcher/unixODBC/commit/45f501e1be2db6b017cc242c79bfb9de32b332a1
Author: Florian Weimer <fweimer@redhat.com>
Last-Update: 2026-01-28

--- a/Drivers/Postgre7.1/info.c
+++ b/Drivers/Postgre7.1/info.c
@@ -1779,14 +1779,14 @@ char index_name[MAX_INFO_STRING];
 char index_name[MAX_INFO_STRING];
 short fields_vector[8];
 char isunique[10], isclustered[10];
-SDWORD index_name_len, fields_vector_len;
+SQLLEN index_name_len, fields_vector_len;
 TupleNode *row;
 int i;
 HSTMT hcol_stmt;
 StatementClass *col_stmt, *indx_stmt;
 char column_name[MAX_INFO_STRING], relhasrules[MAX_INFO_STRING];
 char **column_names = 0;
-Int4 column_name_len;
+SQLLEN column_name_len;
 int total_columns = 0;
 char error = TRUE;
 ConnInfo *ci;
@@ -2136,7 +2136,7 @@ StatementClass *tbl_stmt;
 StatementClass *tbl_stmt;
 char tables_query[STD_STATEMENT_LEN];
 char attname[MAX_INFO_STRING];
-SDWORD attname_len;
+SQLLEN attname_len;
 char pktab[MAX_TABLE_LEN + 1];
 Int2 result_cols;
